File: s3/buckets/templates/1/lua/code/HelloWorld.lua
assert(table.getn(arg) > 0)

local fin = assert(io.open(arg[1], "r"))
local data = fin:read("*all")
fin:close()

local result = string.format("Hello %s", data)
print(result)

local fout = assert(io.open("../results/result.txt", "w"))
local data = fout:write(result)
fout:close()

File: s3/buckets/templates/1/c/code/helloworld.c
#include <stdio.h>

int main(int argc, char *argv[])
{
    char data[64];
    FILE *infile, *outfile;

    if (argc < 2)
        return -1;

    infile = fopen(argv[1], "r");
    if (infile == NULL)
        return -1;

    if (fgets(data, sizeof(data), infile) != NULL)
        printf("Hello %s", data);

    fclose(infile);

    outfile = fopen("../results/result.txt", "w");
    fprintf(outfile, "Hello %s", data);
    fclose(outfile);

    return 0;
}
